### PR TITLE
Bump `prometheus/common` to v0.55.0

### PR DESCRIPTION
**Pull request by dims**
I was catching up on https://github.com/golang/go/issues/19307 and realized that we have 2 `goautoneg` forks in our dependencies. 

Checking on the one that prometheus forked internally, they ended up standardizing on what we use already namely the one maintained by @munnerz [here](https://github.com/munnerz/goautoneg):
https://github.com/prometheus/common/pull/625

So let's switch to the latest prometheus/common to drop the extra copy.

BONUS: one less transitive dependency on `google.golang.org/appengine` ( see https://github.com/prometheus/common/compare/v0.48.0...v0.55.0 )

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mrueg** [comment]
@dims fyi that there might be a future change coming once we're aligned: https://github.com/prometheus/common/issues/652

**Comment by dims** [comment]
@mrueg ACK!

**Comment by BenTheElder** [comment]
/priority important-longterm
working through the diff ...

**Comment by BenTheElder** [comment]
Got through most of it, github has been really slow to expand the diffs ... will come back to this soon.

File: vendor/github.com/stretchr/testify/mock/mock.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// regex for GCCGO functions
+var gccgoRE = regexp.MustCompile(`\.pN\d+_`)
```
Comment (BenTheElder) [inline vendor/github.com/stretchr/testify/mock/mock.go:22]: not ideal in runtime code but should be fine in a mock package.

Comment (BenTheElder) [reply vendor/github.com/stretchr/testify/mock/mock.go:22]: I guess we don't have any way to ensure it's test-import only? something to investigate in the future.

**Review by BenTheElder — COMMENTED** [review]
noting binary in `vendor/google.golang.org/protobuf/internal/editiondefaults/editions_defaults.binpb`

/lgtm
/approve

**Comment by dims** [comment]
/assign @liggitt @thockin 

(for go.mod/sum changes, needs [staging/src/k8s.io/api/OWNERS](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/OWNERS) approval)

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**